examples/train_model.py

```diff
@@ -128,12 +128,17 @@ def main(parser):
     impatience = 0
     saved = False
     valid_world = None
+    num_epochs = 0
+    sampling_data = opt.get('datatype', 'train') == 'train'
     with world:
         while True:
             world.parley()
             parleys += 1
 
-            if opt['num_epochs'] > 0 and parleys >= max_parleys:
+            if world.epoch_done():
+                num_epochs += 1
+                world.reset()
```

Comment:
do you have to reset when an epoch is done? it doesnt keep going round? i cant remember what we settled on..

Reply:
if the world doesn't reset, then the world in the batch that finished the epoch will not see any more examples (it will continue to return epoch_done as true). In addition, resetting puts the batches back at the beginning of the data, so that they may go through it again

Reply:
reset can be good for resetting the data, but it also resets the metrics, so we need to make sure we get the train metrics first (currently happening on line 157)